src/Compilers/Core/Portable/Diagnostic/Diagnostic_SimpleDiagnostic.cs

```diff
@@ -146,40 +146,6 @@ public override IReadOnlyList<Location> AdditionalLocations
                 get { return _properties; }
             }
 
-            public override bool Equals(Diagnostic obj)
-            {
-                var other = obj as SimpleDiagnostic;
-                if (other == null)
-                {
-                    return false;
-                }
-
-                if (AnalyzerExecutor.IsAnalyzerExceptionDiagnostic(this))
```

Comment:
Does "description field" mean ```Descriptor```? If so, ```GetHashCode``` should check ```IsAnalyzerExceptionDiagnostic``` to ensure the same hash code is returned for analyzer exception diagnostics that compare ```Equals``` with different ```Descriptor```.